pkg/alertmanager/multitenant.go

```diff
@@ -414,15 +414,11 @@ func createMultitenantAlertmanager(cfg *MultitenantAlertmanagerConfig, fallbackC
 			return nil, errors.Wrap(err, "failed to initialize Alertmanager's lifecycler")
 		}
 
-		am.ring, err = ring.NewWithStoreClientAndStrategy(am.cfg.ShardingRing.ToRingConfig(), RingNameForServer, RingKey, ringStore, ring.NewIgnoreUnhealthyInstancesReplicationStrategy())
```

Comment:
Few lines above, the call to `ring.NewBasicLifecycler()` is not wrapping the Prometheus registry to add the `cortex_` prefix. I think it should be fixed.